Monytor.PostgreSQL/Repositories/SeriesRepository.cs

```diff
@@ -0,0 +1,128 @@
+using Monytor.Core.Repositories;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+
+namespace Monytor.PostgreSQL {
+
+
+    public class SeriesRepository : ISeriesRepository {
+
+       
+
+        private readonly IDocumentStore _store;
+
+        public SeriesRepository(IDocumentStore store) {
+            _store = store;
+        }        
+
+        public Dictionary<string, IEnumerable<string>> GetGroupValueSummary() {
+
+            using (var session = _store.QuerySession()) {
+                var result = session.Query<TagGroupResult>(@"SELECT  json_build_object('Group', data->'Group', 'Tag' , data->'Tag') 
+	                                                    FROM public.mt_doc_series
+	                                                    group by data->'Group', data->'Tag'
+	                                                    order by data->'Group', data->'Tag';")
+                                        .ToList();
+
+                return result.GroupBy(g => g.Group)
+                     .ToDictionary(g => g.Key, g => g.Select(x => x.Tag));
+            }
+        }
+
+        public Series GetSeries(int id) {
+            using (var session = _store.OpenSession()) {
+                return session.Load<Series>(id);
+            }
+        }
+
+        public IEnumerable<Series> GetSeries(SeriesQuery queryModel) {
+            using (var session = _store.OpenSession()) {              
+                var query = session.Query<Series>()
+                    .Where(x => x.Time >= queryModel.Start
+                    && x.Time <= queryModel.End
+                    && x.Tag == queryModel.Tag
+                    && x.Group == queryModel.Group);
+
+                if (queryModel.OrderBy == Ordering.Ascending) {
+                    query = query.OrderBy(x => x.Time);
+                }
+                else {
+                    query = query.OrderByDescending(x => x.Time);
+                }
+
+                query = query.Take(queryModel.MaxValues);
+
+                return query;
+            }
+        }
+
+        public IEnumerable<Series> GetSeriesByDayMean(SeriesQuery queryModel) {
+            using (var session = _store.OpenSession()) {
+                var query = session.Query<Series>()
+                    .Where(x => x.MatchesSql("date_trunc('hour',CAST(data->>'Time' as timestamp)) >= ? AND date_trunc('hour',CAST(data->>'Time' as timestamp)) <= ?", queryModel.Start, queryModel.End)                     
+                    && x.Tag == queryModel.Tag
+                    && x.Group == queryModel.Group);
+
+                if (queryModel.OrderBy == Ordering.Ascending) {
+                    query = query.OrderBy(x => x.Time);
+                }
+                else {
+                    query = query.OrderByDescending(x => x.Time);
+                }
+                query = query.Take(queryModel.MaxValues);
+
+                var series = query.ToList();
```

Comment:
No group on server side? Currently this will be executed in memory.

Reply:
Good point, guess I was happy that it worked :) I have updated both queries.